### PR TITLE
fix: allow launching language server on empty files

### PR DESCRIPTION
**Pull request by mjlbach**

### CODE REVIEW

**Comment by mjlbach** [comment]
@clason rough fix

**Comment by clason** [comment]
Thanks, looks good!

File: lua/lspconfig/configs.lua [hunk 1]
```diff
@@ -216,7 +218,13 @@ function configs.__newindex(t, config_name, config_def)
       if root_dir then
         id = manager.add(root_dir, false)
       elseif config.single_file_support then
-        local pseudo_root = util.path.dirname(api.nvim_buf_get_name(0))
+        local bufname = api.nvim_buf_get_name(0)
```
Comment (williamboman) [inline lua/lspconfig/configs.lua:221]: Shouldn't this get the bufname for `bufnr`?

File: lua/lspconfig/configs.lua [hunk 2]
```diff
@@ -216,7 +218,13 @@ function configs.__newindex(t, config_name, config_def)
...
       elseif config.single_file_support then
-        local pseudo_root = util.path.dirname(api.nvim_buf_get_name(0))
+        local bufname = api.nvim_buf_get_name(bufnr)
+        local pseudo_root
+        if bufname ~= "" then
+          pseudo_root = util.path.dirname(bufname)
+        else
+          pseudo_root = vim.loop.cwd()
+        end
         id = manager.add(pseudo_root, true)
```
Comment (clason) [inline lua/lspconfig/configs.lua:228]: @mjlbach I know how you hate `cwd` hackery, but I'd still like to remove the error; I don't really care about the attaching servers to unnamed buffers. Would you consider this PR with a change like the following to be mergable?

```suggestion
        if bufname ~= "" then
          local pseudo_root = util.path.dirname(bufname)
          id = manager.add(pseudo_root, true)
        end
```

**Pull request closed without merging**